### PR TITLE
NMS-13170: add related event view to alarm detail page

### PR DESCRIPTION
**Pull request by RangerRick**
Disabled by default.  Enable by setting
org.opennms.web.console.alarms.relatedEvents=true

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-13170

### CODE REVIEW

**Comment by RangerRick** [comment]
Sample screenshot:

Screen Shot 2021-02-22 at 4 04 40 PM

File: opennms-webapp/src/main/java/org/opennms/web/controller/alarm/AlarmDetailController.java [hunk 1]
```diff
@@ -200,4 +220,80 @@ public ModelAndView saveJournalMemo(HttpServletRequest httpServletRequest, HttpS
...
+        } catch (final Exception e) {
+            logger.error("Could not retrieve events for queryCriteria '{}'.", queryCriteria);
+        }
+        return relatedEvents;
+    }
+
+    private List<Filter> getFilters(final OnmsAlarm alarm, final ServletContext context) {
+        final List<Filter> filters = new ArrayList<>();
+
+        // final AlarmIDFilter alarmIDFilter = new AlarmIDFilter(alarm.getId());
```
Comment (christianpape) [inline opennms-webapp/src/main/java/org/opennms/web/controller/alarm/AlarmDetailController.java:278]: Can this line be deleted?

File: opennms-webapp/src/main/java/org/opennms/web/controller/alarm/AlarmDetailController.java [hunk 2]
```diff
@@ -112,23 +129,26 @@ public ModelAndView detail(HttpServletRequest httpServletRequest, HttpServletRes
...
             throw new AlarmIdNotFoundException("Could not find alarm with ID: " + alarmIdString, alarmIdString);
         }
 
         // return to view WEB-INF/jsp/alarm/detail.jsp
         ModelAndView mv = new ModelAndView("alarm/detail");
-        mv.addObject("alarm", m_alarm);
+        mv.addObject("alarm", alarm);
         mv.addObject("alarmId", alarmIdString);
         mv.addObject("acknowledgments", acknowledgments);
+        if (Boolean.getBoolean("org.opennms.web.console.alarms.relatedEvents")) {
```
Comment (christianpape) [inline opennms-webapp/src/main/java/org/opennms/web/controller/alarm/AlarmDetailController.java:149]: Can you add an entry in `opennms-base-assembly/src/main/filtered/etc/opennms.properties` with a short description and the default value?

**Review by christianpape — CHANGES_REQUESTED** [review]
Only the entry in opennms.properties is missing.

**Comment by j-white** [comment]
Any reason not to enable this by default?

**Comment by RangerRick** [comment]
> Any reason not to enable this by default?

I suppose not, I just wasn't sure anyone other than the customer would care about it.  :)

File: opennms-webapp/src/main/java/org/opennms/web/controller/alarm/AlarmDetailController.java (hunk 1 above)
Comment (RangerRick) [reply opennms-webapp/src/main/java/org/opennms/web/controller/alarm/AlarmDetailController.java:278]: Ah, yeah, that's from when I was testing making the filters.  Good catch.

File: opennms-webapp/src/main/java/org/opennms/web/controller/alarm/AlarmDetailController.java (hunk 2 above)
Comment (RangerRick) [reply opennms-webapp/src/main/java/org/opennms/web/controller/alarm/AlarmDetailController.java:149]: Oops, this branch was supposed to be based off `foundation-2019`, which is why I didn't change `opennms.properties` by default.  I was gonna do that when it merged forward to develop.

**Review by cgorantla — COMMENTED** [review]
Looks good to me.

**Review by cgorantla — APPROVED** [review]

**Comment by RangerRick** [comment]
I'd love to not have the extra junk with properties at all, so if everyone's cool with me just adding this as a feature on `foundation-2019` and up, I'll just have it always on.

**Comment by RangerRick** [comment]
(executive decision, decided to go ahead and just enable it for everyone)

**Review by christianpape — APPROVED** [review]
Looks good to me.

**Pull request merged**